### DISCUSSION TITLE
The type or namespace name 'X' could not be found in unit tests

### DISCUSSION THREAD

**Discussion by ajsaclayan** [dotnet/roslyn#73509] [Q&A] [answered]
Hi all,
I've been following [this tutorial](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix) on creating an analyzer. Specifically, I wanted to add some code suggestions when working with an `Activity`. The analyzer/code fix works, but I'm struggling to get the unit tests to pass:

```
		[TestMethod]
		public async Task NoAddTag_NoDiagnostic()
		{
			await VerifyCS.VerifyAnalyzerAsync(@"
using System.Diagnostics;

class Program
{
    static void Main()
    {
        ActivitySource source = new ActivitySource(""MyCompany.MyProduct.MyLibrary"");
        using (var activity = source.StartActivity(""MyActivity""))
        {
            Console.WriteLine(""test"");
        } 
    }
}

");
		}
```
A little bit of background. `ActivitySource` class is part of of `System.Diagnostics`, but I'm getting the following error:
```
    Diagnostics:
    // /0/Test0.cs(8,9): error CS0246: The type or namespace name 'ActivitySource' could not be found (are you missing a using directive or an assembly reference?)
    DiagnosticResult.CompilerError("CS0246").WithSpan(8, 9, 8, 23).WithArguments("ActivitySource"),
    // /0/Test0.cs(8,37): error CS0246: The type or namespace name 'ActivitySource' could not be found (are you missing a using directive or an assembly reference?)
    DiagnosticResult.CompilerError("CS0246").WithSpan(8, 37, 8, 51).WithArguments("ActivitySource"),
    // /0/Test0.cs(11,13): error CS0103: The name 'Console' does not exist in the current context
    DiagnosticResult.CompilerError("CS0103").WithSpan(11, 13, 11, 20).WithArguments("Console"),
```

I've explicitly added the reference to the test project:
```
...
    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.1" />
...
```
My question is how is there a special way to add dependencies to my unit tests?

**Comment by huoyaoyuan** [comment, marked as answer]
> I've explicitly added the reference to the test project:

This does not work. The reference belongs to the process that *executes the test* ("host"), not the code that's being compiled ("target"). When working with compilers, it's especially important to distinguish host and target. Don't use things from host process for compilation.

You can search for examples of `ReferenceAssemblies = ReferenceAssemblies.Net.Net60` or similar in roslyn repo.

**Reply by ajsaclayan** [reply]
Thanks for the info. I was able to get it working with the suggestion you provided. :)
